File: api/application/src/main/java/net/corda/v5/application/messaging/FlowInfo.java
package net.corda.v5.application.messaging;

import net.corda.v5.base.annotations.CordaSerializable;
import org.jetbrains.annotations.NotNull;

@CordaSerializable
public interface FlowInfo {
    /**
     * @return The protocol name of the flow is running.
     */
    @NotNull
    String protocol();

    /**
     * @return The protocol version the flow is running.
     */
    @NotNull
    Integer protocolVersion();
}

File: application/src/main/kotlin/net/corda/v5/application/messaging/FlowInfo.kt
package net.corda.v5.application.messaging

import net.corda.v5.base.annotations.CordaSerializable

/**
 * Version and name of the CorDapp hosting the other side of the flow.
 */
@CordaSerializable
data class FlowInfo(
        /**
         * The integer flow version the other side is using.
         * @see InitiatingFlow
         */
        val flowVersion: Int,
        /**
         * Name of the CorDapp jar hosting the flow, without the .jar extension. It will include a unique identifier
         * to deduplicate it from other releases of the same CorDapp, typically a version string. See the
         * [CorDapp JAR format](https://docs.corda.net/cordapp-build-systems.html#cordapp-jar-format) for more details.
         */
        val appName: String)
